core/src/main/java/org/infinispan/container/offheap/UnpooledOffHeapMemoryAllocator.java

```diff
@@ -44,16 +45,27 @@ public void deallocate(long memoryAddress) {
 
    @Override
    public void deallocate(long memoryAddress, long size) {
-      long currentSize = amountAllocated.addAndGet(- size);
+      long alignedMemoryLength = alignTo8(size);
+      amountAllocated.add(- alignedMemoryLength);
       if (trace) {
-         log.tracef("Deallocating off heap memory at 0x%016x with %d bytes. Total size: %d", memoryAddress, size,
-               currentSize);
+         log.tracef("Deallocating off heap memory at 0x%016x with %d bytes. Total size: %d", memoryAddress,
+               alignedMemoryLength, amountAllocated.sum());
       }
       MEMORY.free(memoryAddress);
    }
 
    @Override
    public long getAllocatedAmount() {
-      return amountAllocated.get();
+      return amountAllocated.sum();
+   }
+
+   /**
+    * Returns a long that is aligned to 8 as most memory allocators will align to this size to more appropriately
```

Comment:
The memory allocator does not change alignment based on the requested size. In C you'd have to use `posix_memalign` to get an alignment different than 8.

Reply:
I wasn't saying it would change based on size. This is just to account for the fact that the memory is aligned when we are counting.

Reply:
Maybe you should just rename the method to `roundUpTo8`? 

Alignment doesn't make sense for longs anyway, so `Round up the size to a multiple of 8 to account for most memory allocators aligning allocations to a multiple of 8` looks clearer to me.

Reply:
Sure